File: staffapp/src/main/java/com/example/staffapp/StaffSessionStore.kt
package com.example.staffapp

import android.content.Context

class StaffSessionStore(context: Context) {
    private val prefs = context.getSharedPreferences("staff_session", Context.MODE_PRIVATE)

    fun saveSession(session: StaffSession) {
        prefs.edit()
            .putString("access", session.accessToken)
            .putString("refresh", session.refreshToken)
            .putString("email", session.userEmail)
            .apply()
    }

    fun loadSession(): StaffSession? {
        val access = prefs.getString("access", null) ?: return null
        val refresh = prefs.getString("refresh", null) ?: return null
        val email = prefs.getString("email", "") ?: ""
        return StaffSession(access, refresh, email)
    }

    fun saveConfig(config: RoleConfig) {
        prefs.edit()
            .putString("roles", config.roles.joinToString(","))
            .putString("app_sections", config.appSections.joinToString(","))
            .putString("admin_sections", config.adminSections.joinToString(","))
            .putString("admin_actions", config.adminActions.joinToString(","))
            .apply()
    }

    fun loadConfig(): RoleConfig? {
        val rolesRaw = prefs.getString("roles", null) ?: return null
        if (rolesRaw.isBlank()) return null
        return RoleConfig(
            roles = parseCsv(rolesRaw),
            appSections = parseCsv(prefs.getString("app_sections", "")),
            adminSections = parseCsv(prefs.getString("admin_sections", "")),
            adminActions = parseCsv(prefs.getString("admin_actions", "")),
            featureFlags = emptyMap(),
        )
    }

    fun clearConfig() {
        prefs.edit()
            .remove("roles")
            .remove("app_sections")
            .remove("admin_sections")
            .remove("admin_actions")
            .apply()
    }

    fun clearAll() {
        prefs.edit().clear().apply()
    }

    fun getLastUnreadNotificationCount(): Int = prefs.getInt("unread_notifications", -1)

    fun setLastUnreadNotificationCount(count: Int) {
        prefs.edit().putInt("unread_notifications", count).apply()
    }

    fun getOrCreatePushToken(): String {
        val existing = prefs.getString("push_token", null)
        if (!existing.isNullOrBlank()) return existing
        val token = "staff-local-${java.util.UUID.randomUUID()}"
        prefs.edit().putString("push_token", token).apply()
        return token
    }

    private fun parseCsv(raw: String?): List<String> {
        return raw.orEmpty().split(",").map { it.trim() }.filter { it.isNotBlank() }
    }
}

File: MatskocaAdmin/MatskocaAdmin/Services/StaffSessionStore.swift
import Foundation

final class StaffSessionStore {
    private let defaults = UserDefaults.standard
    private let suiteKey = "staff_session"

    func saveSession(_ session: StaffSession) {
        defaults.set(session.accessToken, forKey: "\(suiteKey).access")
        defaults.set(session.refreshToken, forKey: "\(suiteKey).refresh")
        defaults.set(session.userEmail, forKey: "\(suiteKey).email")
    }

    func loadSession() -> StaffSession? {
        guard let access = defaults.string(forKey: "\(suiteKey).access"),
              let refresh = defaults.string(forKey: "\(suiteKey).refresh") else {
            return nil
        }
        let email = defaults.string(forKey: "\(suiteKey).email") ?? ""
        return StaffSession(accessToken: access, refreshToken: refresh, userEmail: email)
    }

    func saveConfig(_ config: RoleConfig) {
        defaults.set(config.roles.joined(separator: ","), forKey: "\(suiteKey).roles")
        defaults.set(config.appSections.joined(separator: ","), forKey: "\(suiteKey).app_sections")
        defaults.set(config.adminSections.joined(separator: ","), forKey: "\(suiteKey).admin_sections")
        defaults.set(config.adminActions.joined(separator: ","), forKey: "\(suiteKey).admin_actions")
    }

    func loadConfig() -> RoleConfig? {
        guard let rolesRaw = defaults.string(forKey: "\(suiteKey).roles"), !rolesRaw.isEmpty else {
            return nil
        }
        return RoleConfig(
            roles: parseCsv(rolesRaw),
            appSections: parseCsv(defaults.string(forKey: "\(suiteKey).app_sections")),
            adminSections: parseCsv(defaults.string(forKey: "\(suiteKey).admin_sections")),
            adminActions: parseCsv(defaults.string(forKey: "\(suiteKey).admin_actions")),
            featureFlags: [:]
        )
    }

    func clearConfig() {
        defaults.removeObject(forKey: "\(suiteKey).roles")
        defaults.removeObject(forKey: "\(suiteKey).app_sections")
        defaults.removeObject(forKey: "\(suiteKey).admin_sections")
        defaults.removeObject(forKey: "\(suiteKey).admin_actions")
    }

    func clearAll() {
        let keys = [
            "\(suiteKey).access", "\(suiteKey).refresh", "\(suiteKey).email",
            "\(suiteKey).roles", "\(suiteKey).app_sections", "\(suiteKey).admin_sections",
            "\(suiteKey).admin_actions", "\(suiteKey).unread_notifications", "\(suiteKey).push_token",
        ]
        keys.forEach { defaults.removeObject(forKey: $0) }
    }

    func getLastUnreadNotificationCount() -> Int {
        defaults.object(forKey: "\(suiteKey).unread_notifications") as? Int ?? -1
    }

    func setLastUnreadNotificationCount(_ count: Int) {
        defaults.set(count, forKey: "\(suiteKey).unread_notifications")
    }

    func getOrCreatePushToken() -> String {
        if let existing = defaults.string(forKey: "\(suiteKey).push_token"), !existing.isEmpty {
            return existing
        }
        let token = "staff-local-\(UUID().uuidString)"
        defaults.set(token, forKey: "\(suiteKey).push_token")
        return token
    }

    private func parseCsv(_ raw: String?) -> [String] {
        guard let raw, !raw.isEmpty else { return [] }
        return raw.split(separator: ",").map { $0.trimmingCharacters(in: .whitespaces) }.filter { !$0.isEmpty }
    }
}
